text/0000-cargo-tags.md

```diff
@@ -0,0 +1,126 @@
+
+1. Instead of being gatekept by a PR to the crates.io repository, categories can organically be adopted by community members in the form of keywords. Which keywords are most popular and useful can be decided organically.
+2. Keywords can still be given descriptions and other metadata on crates.io, although no distinction between these "special" keywords and other keywords is made in cargo itself. This allows making changes to the way crates are presented without having to worry about backwards compatibility.
+3. Adding, removing, and modifying the curated set of keywords is no longer a technical choice, but a cultural one.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Categories and keywords in `Cargo.toml` are now deprecated. Setting `categories` or `keywords` will now trigger a warning and suggest to use `tags` instead. These combined `tags` now accept the formats used by both `categories` and `keywords`, meaning:
+
+* It's now possible to add a single double-colon inside a tag to indicate a parent tag. This has the effect of adding two tags to a crate: for example, adding the `development-tools::testing` tag adds the crate to both the `development-tools` tag and the `development-tools::testing` tag. The part after the double-colon is called a subtag.
+* On each side of the double-colon, the length of text may be up to 25 characters. That means that, including the double-colon, tags can be up to 52 characters long. (This is to accomodate the accomodate the largest category before the unification, `development-tools::procedural-macro-helpers`, although rounding up to a nice number for the actual limit.)
+
+On crates.io, the "categories" section of the sidebar is removed and all tags are shown with hashtags in a crate header, like keywords are now. On the pages for popular tags, a curated description from the crates.io team may be shown alongside a list of common subtags and similar tags.
+
+The exact policy for how this curated metadata can be added to tags can be modified by the crates.io team without an RFC, although an initial policy is described below in the reference-level explanation.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The `Cargo.toml` manifest gains a new key, `package.tags`, which is an array of strings with the following restrictions:
+
+* The strings are nonempty and consist of "tag components" separated by double-colons as specified below. While crates.io limits the number of components to at most two, this can be raised for custom registries.
+* Tag components must start with a letter and only contain letters, numbers, `_`, or `-`. (This is the same as keywords and crate names.)
```

Comment:
what about `3d`?

Reply:
`3d` is currently allowed as a keyword, so I'm assuming the "This is the same as keywords and crate names" applies and the rule is just slightly incorrect :)

Reply:
Yeah, I was basing this off the docs for Cargo, which appear to be inaccurate: https://doc.rust-lang.org/cargo/reference/manifest.html#the-keywords-field

At least how it is at the time of writing, what's documented and what's allowed don't match.

Reply:
(If I can get to it this weekend, I'll poke around the code and see what the actual current requirements are and submit a PR to update the docs.)

Reply:
the short summary from memory is:
- starts with alphanumeric
- continues with alphanumeric, `_`, `-`, or `+`

Reply:
@Turbo87 updated the docs and I updated the RFC!